app/components/improvedHome/index.tsx

```diff
@@ -136,10 +139,37 @@ const ImprovedHome: React.FC<Props> = props => {
   React.useEffect(
     () => {
       if (currentUser.isLoggedIn) {
-        Promise.all([dispatch(fetchBasedOnActivityPapers()), dispatch(fetchBasedOnCollectionPapers())]);
+        setIsLoadingBasedOnActivityPapers(true);
+        setIsLoadingBasedOnCollectionPapers(true);
+
+        const fetchBasedOnActivityPapers = homeAPI
+          .getBasedOnActivityPapers()
+          .then(res => {
+            setBasedOnActivityPapers(res);
+            setIsLoadingBasedOnActivityPapers(false);
+          })
+          .catch(err => {
+            console.error(err);
+            setIsLoadingBasedOnActivityPapers(false);
+          });
+
+        const fetchBasedOnCollectionPapers = homeAPI
+          .getBasedOnCollectionPapers()
+          .then(res => {
+            setBasedOnCollectionPapers(res);
+            setIsLoadingBasedOnCollectionPapers(false);
+          })
+          .catch(err => {
+            console.error(err);
+            setIsLoadingBasedOnCollectionPapers(false);
+          });
+
+        Promise.all([fetchBasedOnActivityPapers, fetchBasedOnCollectionPapers]);
```

Comment:
you don't need to wait for these promises.

Reply:
resolved